### PR TITLE
feat: add logicalId validation [sc-0]

### LINKED ISSUE #487: story: Restrict logical id to alphanumeric with special characters [a-zA-Z0-9_-/#]
### Story description

We want to help our users avoid using ids prone to making mistakes, like empty spaces or tabs. I suggest using `[a-zA-Z0-9_-/#]` or simply `[\w-/#]` to make sure they have a simpler but still extensive logical ids combination. This should be a FE and a BE restriction

### Shortcut link

_No response_

### Additional resources

_No response_

### PR DESCRIPTION
**Pull request by nahuelon**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [ ] Create CLI
* [ ] Test
* [ ] Docs
* [ ] Examples
* [ ] Other

## Notes for the Reviewer
- Add `logicalId` validation.
- Now the `Project` construct class extends from `Construct`

> Resolves #487

## New Dependency Submission
- Important: we have to confirm all Production/Staging projects compliant with the new `logicalId` format restriction.

### CODE REVIEW

File: packages/cli/src/constructs/construct.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import { Session } from './project'
 import { Ref } from './ref'
+import { ValidationError } from './validator-error'
 
 export abstract class Construct {
   type: string
   logicalId: string
   constructor (type: string, logicalId: string) {
+    if (!/^[A-Za-z0-9_\-/#.]+$/.test(logicalId)) {
```
Comment (umutuzgur) [inline packages/cli/src/constructs/construct.ts:9]: Any reason to add `.` support as well? Did you find anyone using it?

Comment (umutuzgur) [reply packages/cli/src/constructs/construct.ts:9]: I just saw your comment in the issue, now I got it 👍

**Comment by umutuzgur** [comment]
Could you also add it here @nahuelon https://github.com/checkly/checkly-cli/blob/main/packages/cli/src/constructs/project.ts#L35-L43?

**Comment by umutuzgur** [comment]
Can we just a create helper that checks the logical id and verifies it in both classes? We can also remove project from the constructs folder as the user can't do anything with it atm

**Comment by nahuelon** [comment]
@umutuzgur , we don't need a helper since the `logicalId` validation is centralized in one place (for all resource contructs including the `Project`). To remove the `project` from the constructs folder we need a small refactor.

File: packages/cli/src/constructs/project.ts [hunk 2]
```diff
@@ -26,13 +26,16 @@ export class Project {
...
   }
 
+  static readonly __checklyType = 'project'
+
   /**
-   * Constructs the Email Alert Channel instance
+   * Constructs the Project instance
    *
-   * @param logicalId unique project name
+   * @param logicalId unique project identificator
```
Comment (umutuzgur) [inline packages/cli/src/constructs/project.ts:34]: identifier

**Comment by umutuzgur** [comment]
I suggested that because Project is not really a construct. However, the solution solves the problem so I'm cool with it

**Review by umutuzgur — DISMISSED** [review]

**Review by umutuzgur — APPROVED** [review]

**Pull request merged**